cabal-install/Distribution/Client/ProjectOrchestration.hs

```diff
@@ -223,8 +223,13 @@ data ProjectBuildContext = ProjectBuildContext {
 runProjectPreBuildPhase
     :: Verbosity
     -> ProjectBaseContext
-    -> (ElaboratedInstallPlan -> IO ElaboratedInstallPlan)
-    -> IO ProjectBuildContext
+    -- ^ A function that selects a subset of the plan and returns it,
+    -- It can also return some arbitrary data, which can be useful in case
+    -- the caller needs it.
+    -> (ElaboratedInstallPlan -> IO (ElaboratedInstallPlan, a))
```

Comment:
Anyhow, I still think that sticking `TargetMap` into `ProjectBuildContext` makes sense.